### PR TITLE
Replacing the utf8 scrubbing with the scrub_rb shim

### PR DESCRIPTION
**Pull request by fneves**
Replacing the `to_utf8!` with the shim.

/cc @ggrossman  @grosser 

### References
 - Jira link: 

### Risks
 - None

### CODE REVIEW

**Comment by fneves** [comment]
We could drop the support for ruby < 2.1, but with this, do we really need to?

**Comment by grosser** [comment]
1 less dependency is always nice ... and the old code did not have any bugs, so it will continue working for older rubies ...

:+1: either way ...

File: lib/credit_card_sanitizer.rb [hunk 1]
```diff
@@ -59,7 +60,7 @@ def initialize(options = {})
   # Returns a String of the redacted text if a credit card number was detected.
   # Returns nil if no credit card numbers were detected.
   def sanitize!(text)
-    to_utf8!(text)
+    text.scrub!('�')
```
Comment (ggrossman) [inline lib/credit_card_sanitizer.rb]: Is it necessary to explicitly specify the replacement char? It looks like [scrub_rb  defaults to it](https://github.com/jrochkind/scrub_rb/blob/master/lib/scrub_rb.rb#L15-L23)

**Comment by ggrossman** [comment]
:+1:

File: lib/credit_card_sanitizer.rb (hunk 1 above)
Comment (fneves) [inline lib/credit_card_sanitizer.rb]: They are not the same. This gem is using the replacement char (unicode 0xFFFD), while scrub_rb was using `?` (unicode 0x3F)

File: lib/credit_card_sanitizer.rb [hunk 2]
```diff
@@ -131,19 +132,4 @@ def without_expiration(text)
     yield
     text.gsub!(expiration_date_boundary, '')
   end
-
-  if ''.respond_to?(:scrub)
-    def to_utf8!(str)
-      str.force_encoding(Encoding::UTF_8)
```
Comment (staugaard) [inline lib/credit_card_sanitizer.rb]: I think we still need to `force_encoding`.

I'm getting these exceptions in Inbox:
`Encoding::CompatibilityError:         Encoding::CompatibilityError: incompatible character encodings: US-ASCII and UTF-8`

Comment (paradox460) [inline lib/credit_card_sanitizer.rb]: We're seeing those same exceptions in billing too

Comment (grosser) [inline lib/credit_card_sanitizer.rb]: yes, just ran into the same in classic ... does not change the encoding ... and if the encoding is binary then it is a noop ...

**Comment by fneves** [comment]
Do we have any input examples that cause those exceptions?
Should we revert this for the time being?

File: lib/credit_card_sanitizer.rb (hunk 2 above)
Comment (fneves) [inline lib/credit_card_sanitizer.rb]: Do we have any input examples that cause those exceptions?
Should we revert this for the time being?

Comment (grosser) [inline lib/credit_card_sanitizer.rb]: I'd say change to:

```
text.force_encoding Encoding::UTF_8
text.scrub!('�')
```

**Comment by ggrossman** [comment]
I think using the `scrub_rb` shim is still the right direction
I'm with @grosser , let's just make sure we're doing the `force_encoding`

**Pull request merged**